pkg/provider/wif_configs.go

```diff
@@ -0,0 +1,56 @@
+package provider
+
+import (
+	"fmt"
+
+	"github.com/openshift-online/ocm-cli/pkg/arguments"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+)
+
+func getWifConfigs(client *cmv1.Client) (wifConfigs []*cmv1.WifConfig, err error) {
+	collection := client.GCP().WifConfigs()
+	page := 1
+	size := 100
+	for {
+		var response *cmv1.WifConfigsListResponse
+		response, err = collection.List().
+			Page(page).
+			Size(size).
+			Send()
+		if err != nil {
+			return
+		}
+		wifConfigs = append(wifConfigs, response.Items().Slice()...)
+		if response.Size() < size {
+			break
+		}
+		page++
```

Comment:
This is a silly edge case, but if the total happens to be equal to the size, then a call to page 2 will needlessly be made.  I think that the make the correct number of list calls would be to check if the accumulated number of resources seen so far is less then the total listed by the call. I think the way that is being done here is uniform with the rest of the codebase, so there's no need to fix it here.

Reply:
monkey see monkey do